test/e2e/ipfs-companion.test.js

```diff
@@ -0,0 +1,176 @@
+    console.info(`Using remote webdriver: ${process.env.SELENIUM_REMOTE_FIREFOX_URL}`)
+    builder.usingServer(process.env.SELENIUM_REMOTE_FIREFOX_URL)
+  }
+  console.info('Starting Firefox')
+  const browser = await builder.build()
+  if (extension !== undefined) {
+    console.info('Installing the extension')
+    await browser.installAddon(extension, true)
+    await delay(5000) // waiting for the extension to load
+  }
+  console.info('Firefox is ready')
+  return browser
+}
+
+const ExtensionURLRegex = /^(moz|chrome)-extension:\/\/[^\/]+/ // eslint-disable-line no-useless-escape
+async function findExtensionUrl (browser) {
+  console.info('Looking for an open extension tab')
+  const handles = await browser.getAllWindowHandles()
+  console.info(`Found ${handles.length} candidates`)
+
+  for (const handle of handles) {
+    console.info('Switching tabs')
+    await browser.switchTo().window(handle)
+    const url = await browser.getCurrentUrl()
+    console.info(`The current URL is: ${url}`)
+    const extensionURL = ExtensionURLRegex.exec(url)?.at(0)
+    if (extensionURL !== undefined) {
+      console.info(`Found the extension URL: ${extensionURL}`)
+      return extensionURL
+    }
+  }
+  console.warn('No extension URL found')
+}
+
+async function updateExtensionSettings (browser, url, id, value) {
+  console.info('Updating extension setting')
+  console.info(`Going to: ${url}/dist/options/options.html`)
+  await browser.get(`${url}/dist/options/options.html`)
+  console.info(`Looking for an element: ${id}`)
+  let element = browser.findElement(By.id(id))
+  console.info(`Setting new value to: ${value}`)
+  await element.sendKeys('')
+  await delay(1000) // waiting for focus to be acquired
+  await element.clear()
+  await delay(1000) // waiting for input to be cleared
+  await element.sendKeys(value)
+  await element.sendKeys(Key.TAB)
+  await delay(5000) // waiting for the setting to be applied
+  console.info('Checking if the update worked')
+  element = browser.findElement(By.id(id))
+  const v = await element.getAttribute('value')
+  console.info('The setting update is complete')
+  return v
+}
+
+async function getNumberOfConnectedPeers (browser, url) {
+  console.info('Checking the number of connected peers')
+  console.info(`Going to: ${url}/dist/landing-pages/welcome/index.html`)
+  await browser.get(`${url}/dist/landing-pages/welcome/index.html`)
+  await delay(5000) // waiting for the connection number to appear
+  if (process.env.TEST_DEBUG === 'true') {
+    const html = await browser.getPageSource()
+    console.debug(html)
+  }
+  console.info('Looking for an element with text: \'Your node is connected to ...\'')
+  const p = browser.findElement(By.xpath("//p[text()='Your node is connected to ']"))
+  const span = p.findElement(By.css('span'))
+  const peers = await span.getText()
+  console.info(`There are ${peers} connected peers`)
+  return parseInt(peers)
+}
+
+async function runTest (browserName) {
```

Comment:
nit: 
```suggestion
async function runBasicBrowserTest (browserName) {
```

non-nit comment: I feel like this test is set up in a way that all of the tests are going to be added here, which we don't want. We can tackle this later.